Parsers/ASimProcessEvent/Parsers/vimProcessCreateVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,273 @@
+              and eventType_s == "endpoint.event.procstart"
+              and (eventtype == '*' or eventtype == 'ProcessCreated')
+              and (array_length(dvcipaddr_has_any_prefix) == 0 or has_any_ipv4_prefix(device_external_ip_s, dvcipaddr_has_any_prefix))
+              and (targetusername_has == '*' or childproc_username_s has targetusername_has)  
+              and (array_length(commandline_has_all) == 0 or target_cmdline_s has_all (commandline_has_all)) 
+              and (array_length(commandline_has_any) == 0 or target_cmdline_s has_any (commandline_has_any)) 
+              and (array_length(commandline_has_any_ip_prefix) == 0 or has_any_ipv4_prefix(target_cmdline_s, commandline_has_any_ip_prefix)) 
+              and (array_length(actingprocess_has_any) == 0 or process_path_s has_any (actingprocess_has_any)) 
+              and (array_length(targetprocess_has_any) == 0 or childproc_name_s has_any (targetprocess_has_any)) 
+              and (array_length(parentprocess_has_any) == 0 or parent_path_s has_any (parentprocess_has_any))
+              and (array_length(dvchostname_has_any) == 0 or device_name_s has_any (dvchostname_has_any))
+              and array_length(hashes_has_any) == 0 or childproc_hash_s has_any (hashes_has_any)
+          | parse process_hash_s with * '["' ActingProcessMD5: string '","' ActingProcessSHA256: string '"]'
+          | parse parent_hash_s with * '["' ParentProcessMD5: string  '","' ParentProcessSHA256: string '"]'
+          | parse childproc_hash_s with * '["' TargetProcessMD5: string '","' TargetProcessSHA256: string '"]'
+          | lookup EventFieldsLookup on sensor_action_s;  
+      let threatsdata = CarbonBlackNotifications_CL
+          | where type_s == "THREAT"
+          | project
+              threatInfo_incidentId_g,
+              threatInfo_score_d,
+              threatInfo_summary_s,
+              threatInfo_time_d,
+              threatInfo_threatCause_threatCategory_s,
+              threatInfo_threatCause_causeEventId_g,
+              ruleName_s,
+              deviceInfo_deviceVersion_s,
+              threatInfo_threatCause_originSourceType_s,
+              threatInfo_threatCause_reputation_s,
+              threatInfo_threatCause_reason_s;
+      let alertsdata = CarbonBlackNotifications_CL
+          | where type_s == "CB_ANALYTICS"
+          | project
+              id_g,
+              primary_event_id_g,
+              deviceInfo_deviceVersion_s,
+              threat_id_g,
+              threatInfo_score_d,
+              threatInfo_summary_s,
+              threatInfo_threatCause_reason_s;
+      let processdatawiththreat = processdata
+          | where isnotempty(alert_id_g) and isnotempty(event_id_g)
+          | lookup threatsdata
+              on
+              $left.alert_id_g == $right.threatInfo_incidentId_g,
+              $left.event_id_g == $right.threatInfo_threatCause_causeEventId_g
```

Comment:
Do we really need to merge these two tables info? Does CarbonBlackNotifications_CL doesn't have other basic process fields like Process , device and user details? I think it has other fields too.
If we are getting other fields in threat logs including entities involved like user, device and activity info we don't need to join the two tables and just do union of them.

If theat logs are missing other critical fields than I think Join might be better in terms of performance instead of lookup. Please test and compare?

Reply:
Currently we have replaced lookup with join as it gives better performance for CPU usage.
Also we have checked that Threat logs contains other fields but it doesn't have ProcessId of TargetProcess.